src/Transpile.ts

```diff
@@ -57,7 +61,11 @@ export function transpile({
     };
 
     if (options.noEmitOnError) {
-        const preEmitDiagnostics = [...program.getOptionsDiagnostics(), ...program.getGlobalDiagnostics()];
+        const preEmitDiagnostics = [
+            ...diagnostics,
```

Comment:
At this point it's always empty

Reply:
What is? diagnostics can have diagnostics from validateOptions at line 51.